spyder/api/plugins.py

```diff
@@ -282,6 +284,18 @@ def get_plugin_icon(self):
         """
         return ima.icon('outline_explorer')
 
+    def set_plugin_icon_size(self, iconsize):
+        """
+        Set the icon size of plugin.
+
+        This must be reimplemented by plugins that need to adjust the icon
+        size of their toolbar.
```

Comment:
@jnsebgosselin, question

Since we are making things uniform here

>This must be reimplemented by plugins that need to adjust the icon size of their toolbar.

Is this docstring still valid? I mean we are handling this for other developers, no?

Reply:
In this PR, I tried to make the creation of the toolbars and main layout uniform.

However, the layout of the plugins encompass a wide range of complexities and more often than not, the plugin themselves are not aware of the toolbar, because it is added deeper in the layout.

Moreover, some plugins have tab bar, other have find widget(s), other have a stack widget. So I think it is better to let the developers some flexibility and let them handle this? Otherwise, we would have to cover a lot of possibilities.

I've implemented the `set_plugin_icon_size` for all the default plugins in this PR. You can take a look and if you think we should handle this all here instead, I can try to find a way.